compaction/incremental_compaction_strategy.cc

```diff
@@ -417,11 +414,10 @@ int64_t incremental_compaction_strategy::estimated_pending_compactions(table_sta
 
 std::vector<shared_sstable>
 incremental_compaction_strategy::runs_to_sstables(std::vector<frozen_sstable_run> runs) {
-    return boost::accumulate(runs, std::vector<shared_sstable>(), [&] (std::vector<shared_sstable>&& v, const frozen_sstable_run& run_ptr) {
-        auto& run = *run_ptr;
-        v.insert(v.end(), run.all().begin(), run.all().end());
-        return std::move(v);
-    });
+    return runs
+        | std::views::transform([] (auto& run) { return run->all(); })
```

Comment:
Won't this create a copy of `all()`? Maybe add `-> const std::vector<shared_sstable>&` to ensure we don't copy.

Reply:
thank you for pointing this out! indeed. i made the same mistake as i did in https://github.com/scylladb/scylladb/pull/22731#discussion_r1944560380.